pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamLevelConsumer.java

```diff
@@ -47,10 +48,9 @@ void start()
   /**
    * Get next row from the stream and decode it into a generic row
    * @param destination the decoded generic row
-   * @param metadata the metadata associated with the decoded stream if any
    * @return the decoded generic row
    */
-  default GenericRow next(GenericRow destination, StreamMessageMetadata metadata) {
+  default GenericRow next(GenericRow destination, RowMetadata metadata) {
```

Comment:
Please remove this API since it's not used right now. This interface is public and stable, so let's be careful about it.

Reply:
I wanted plumb-in the implementation for both HLC and LLC - so its less confusing in the future. Given there is no specific use case for this now, I will drop it - once things look good with LLC, I might update the interface to make both support similar functionality.